### DISCUSSION TITLE
Renovate bot can't always automerge in monorepo with codeowners rules applied

### DISCUSSION THREAD

**Discussion by MDietz135** [renovatebot/renovate#19820] [Request Help] [answered]
Hi there,

we're using renovate bot to maintain our dependencies within our monorepo. The monorepo is managed with turborepo. 

If the dependency which needs to be updated is only located within a single `package.json` file (e.g. the root level one) renovate bot is able to override the merge request approval settings and will automerge the MR. 

However, as soon as the dependency is located within two or more `package.json` files (e.g. root level and within a file located in a sub package of the monorepo) renovate bot is no longer able to override the merge request approval settings and the MR needs to be merged manually. Since I was not able to find a similar discussion I openend this one.

Is it possible to configure renovate bot in a way that the approval rules are overridden for every MR? Both package.json files (the root level one as well as the one located in a sub package of the monorepo) have codeowners applied.

This is the renovate config we're currently using:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "baseBranches": ["master"],
  "extends": [
    "config:base",
    ":semanticCommits",
    "docker:enableMajor",
    ":disableDependencyDashboard"
  ],
  "updateNotScheduled": false,
  "automergeSchedule": ["after 10pm every weekday", "before 5am every weekday", "every weekend"],
  "schedule": ["after 10pm every weekday", "before 5am every weekday", "every weekend"],
  "timezone": "Europe/Berlin",
  "kubernetes": {
    "fileMatch": [".\\.gitlab-ci\\.yml$"]
  },
  "gitlabci-include": {
    "fileMatch": [".\\.gitlab-ci\\.yml$", "^.gitlab-ci/.*\\.yml$", "^.gitlab-ci/jobs/.*\\.yml$"]
  },
  "packageRules": [
    {
      "extends": ["monorepo:material-ui", "monorepo:emotion"],
      "groupName": "material-ui and emotion"
    },
    {
      "matchPackagePatterns": ["storybook"],
      "groupName": "storybook dependencies"
    },
    {
      "matchPackagePatterns": ["^next", "@next"],
      "groupName": "nextjs dependencies"
    },
    {
      "matchPackagePatterns": ["embla-carousel"],
      "groupName": "embla carousel dependencies"
    },
    {
      "matchPackagePatterns": ["lint", "prettier", "eslint"],
      "matchPackageNames": ["eslint-import-resolver-typescript"],
      "groupName": "eslint and prettier"
    },
    {
      "extends": ["packages:react"],
      "matchPackagePatterns": ["^@types/react"],
      "groupName": "react dependencies"
    },
    {
      "extends": ["packages:jsUnitTest"],
      "groupName": "unit testing dependencies"
    },
    {
      "matchPackageNames": ["npm", "node"],
      "groupName": "node dependencies"
    },
    {
      "description": "Trigger breaking release for major updates",
      "matchUpdateTypes": ["major"],
      "semanticCommitType": "feat",
      "commitBody": "BREAKING CHANGE: Major update"
    },
    {
      "description": "Trigger feature release for minor updates",
      "matchUpdateTypes": ["minor"],
      "semanticCommitType": "feat"
    },
    {
      "description": "Trigger fix release for patch updates",
      "matchUpdateTypes": ["patch", "digest"],
      "semanticCommitType": "fix"
    },
    {
      "updateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true,
      "gitLabAutomerge": true,
      "gitLabIgnoreApprovals": true
    },
    {
      "matchPackagePatterns": ["*"],
      "rangeStrategy": "bump"
    }
  ],
  "regexManagers": [
    {
      "fileMatch": ["(^|/|\\.)Dockerfile$", "(^|/)Dockerfile\\.[^/]*$"],
      "matchStrings": [
        "# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s"
      ],
      "versioningTemplate": "{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"
    }
  ],
  "prHourlyLimit": 10,
  "prConcurrentLimit": 5,
  "dependencyDashboard": false,
  "gitLabIgnoreApprovals": true,
  "platformAutomerge": true,
  "automergeType": "pr"
}
```

Thank's for your help in advance

**Comment by rarkins** [comment, marked as answer]
Do you mean GitLab assigns mandatory reviewers based on codeowners?

If so then your question is whether either is possible:
- can you disable this gitlab feature of assignment?
- can you bypass this assignment/requirement?

**Reply by MDietz135** [reply]
If the dependency is located in only one package.json file renovate is able to remove the merge request approval rules - if the dependency is located within multiple package.json files approval rules are not removed from the merge request. In both cases codeowners are applied to the package.json files and, initally, also to the MR. 

So I am a bit confused why, in one case, the aproval rules can be removed an in an other case the can't be removed. 

Currently I am not able to disable the required codeowners approval for merge requests within our project since it is a mandatory part within our code review/release process within our company.

**Reply by MDietz135** [reply]
Okay - forget it. Everything works as expected, it is a configuration issue within our repository. Sorry for the inconvenience and thanks for your quick reply
